clients/telemetry-pubsub/src/main/java/org/eclipse/hono/client/telemetry/pubsub/PubSubBasedDownstreamSender.java

```diff
@@ -115,17 +119,25 @@ public Future<Void> sendEvent(
         final String tenantId = tenant.getTenantId();
         final String deviceId = device.getDeviceId();
 
+        final Map<String, String> propertiesStrings = properties.entrySet().stream()
+                .filter(entry -> entry.getValue() instanceof String)
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> (String) e.getValue()));
```

Comment:
processing all properties just to get the value of the *orig_address* property seems very inefficient. How about
```java
final String orig_address = Optional.ofNullable(properties.get(MessageHelper.APP_PROPERTY_ORIG_ADDRESS))
   .filter(String.class::instanceof)
   .map(String.class::cast)
  .orElse("");
```

and then pass in the address string into PubSubMessageHelper.getSubtopics() instead of the map ...

Reply:
Very good idea, I changed this.